docs/developer-docs/smart-contracts/maintain/settings.mdx

```diff
@@ -117,3 +117,107 @@ async fn modify_canister_settings() -> CallResult<()> {
+
+:::info
+Currently, `memory_allocation` also denotes the maximum amount of memory that the canister is allowed to use in total.
+For example, if a canister has memory allocation of 8GiB, then it pre-allocates 8GiBs and cannot exceed that amount.
+This constraint is considered a mis-design and may be lifted in the future.
+:::
+
+Canisters pay a rent fee for memory allocation meaning that the payment depends on time and the amount of memory allocation regardless whether canister actually uses that memory or not.
+You can learn more about payment for memory [here](docs/current/developer-docs/gas-cost).
+
+The default value of memory allocation is 0 bytes, which means that there is no memory allocation.
+
+## Freezing threshold
+
+If a canister runs out of cycles, then it gets uninstalled, which means that its code and data are deleted.
+Canister retains only its metadata such as the canister id, controllers, settings, etc.
+
+The freezing threshold protects against canister unexpectedly running out of cycles.
+This field denotes time duration in seconds -- the minimum duration that the canister should exist without running out of cycles.
+
+When the cycles balance of the canister becomes too low, then the system freezes the canister in order to guarantee its freezing threshold.
+A frozen canister does not execute messages and pays only for renting resources such as compute allocation, memory allocation, and memory usage.
+
+:::info
+See also the `idle_cycles_burned_per_day` field in canister status to learn how many cycles a canister consumes in its idle or frozen state.
+:::
+
+The default value of the freezing threshold is 2592000 (approximately 30 days).
+
+## Reserved cycles limit
+
+In addition to the main cycles balance, a canister has a secondary cycles balance called reserved cycles.
+The reserved cycles are cycles that are set aside for future resource payments by the resource reservation mechanism.
+You can learn more about resource reservation [here](docs/current/developer-docs/gas-cost).
+
+The `reserved_cycles_limit` field in canister settings serves as the upper limit on reserved cycles.
+It protects against spending too many cycles in resource reservation.
+
+The default value is 5 trillion cycles.
+
+## Wasm memory limit
+
+A canister has a Wasm memory and a stable memory.
+Currently, the Wasm memory is 32-bit, which means that it cannot grow beyond the hard limit of 4GiB.
+
+If a canister stores user data in the Wasm memory instead of the stable memory, then its memory usage will increase with the number of users.
+Eventually, such a canister could reach 4GiB at which point it won’t be able to allocate more memory and will stop working.
+It is also likely that the pre-upgrade hook will fail to allocate, which will make the canister and its data unrecoverable.
+Even if the canister doesn’t store user data in Wasm memory, its memory usage may increase due to memory leaks.
```

Comment:
```suggestion
A canister has a Wasm memory and a stable memory. Currently, the Wasm memory is 32-bit, which means that it cannot grow beyond the hard limit of 4GiB.

If a canister stores user data in the Wasm memory instead of the stable memory, then its memory usage will increase with the number of users. Using that configuration, a canister could reach 4GiB, at which point it won’t be able to allocate more memory and will stop working. It is also likely that the pre-upgrade hook will fail to allocate, which will make the canister and its data unrecoverable.

Even if the canister doesn’t store user data in Wasm memory, its memory usage may increase due to memory leaks.
```